src/cobra/core/reaction.py

```diff
@@ -62,20 +71,30 @@ class Reaction(Object):
     ----------
     id : string
         The identifier to associate with this reaction
-    name : string
+    name : string, optional
         A human readable name for the reaction
-    subsystem : string
+    subsystem : string, optional
         Subsystem where the reaction is meant to occur
-    lower_bound : float
+    lower_bound : float, optional
         The lower flux bound
-    upper_bound : float
+    upper_bound : float, optional
         The upper flux bound
+    **kwargs:
+        Further keyword arguments are passed on to the parent class.
     """
 
+    # noinspection PyShadowingBuiltins
     def __init__(
-        self, id=None, name="", subsystem="", lower_bound=0.0, upper_bound=None
-    ):
-        Object.__init__(self, id, name)
+        self,
+        id: Optional[str] = None,
+        name: Optional[str] = "",
```

Comment:
Please only use `str` for the type annotation.

Reply:
If the default value is `""` (empty string), the type should be `str`. And, if the default value is `None`, then the type becomes `Optional[...]`.